src/main/java/org/springframework/data/elasticsearch/core/convert/ElasticsearchConverter.java

```diff
@@ -72,14 +75,24 @@ default ProjectionFactory getProjectionFactory() {
 	/**
 	 * Map a single {@link Document} to an instance of the given type.
 	 *
-	 * @param document must not be {@literal null}.
+	 * @param document the document to map
 	 * @param type must not be {@literal null}.
 	 * @param <T>
-	 * @return can be {@literal null} if the {@link Document#isEmpty()} is true.
+	 * @return can be {@literal null} if the document is null or {@link Document#isEmpty()} is true.
 	 * @since 4.0
 	 */
 	@Nullable
-	<T> T mapDocument(Document document, Class<T> type);
+	<T> T mapDocument(@Nullable Document document, Class<T> type);
+
+	/**
+	 * builds a {@link SearchHit} from a {@link SearchDocument}.
+	 * 
+	 * @param searchDocument must not be {@literal null}
+	 * @param type the type of the returned data, must not be {@literal null}.
+	 * @param <T> the clazz of the type, must not be {@literal null}.
+	 * @return SearchHIt with all available information filled in
+	 */
+	<T> SearchHit<T> toSearchHit(SearchDocument searchDocument, Class<T> type);
```

Comment:
Let's align with `EntityReader`'s `read` method so we would have `<R> SearchHit<R> read(Class<R> type, SearchDocument searchDocument);`